test/Npgsql.Tests/ReaderTests.cs

```diff
@@ -1228,6 +1228,62 @@ public async Task ReaderReuseOnDispose()
             Assert.That(reader1, Is.SameAs(reader2));
             await reader2.DisposeAsync();
         }
+
+        [Test]
+        public async Task UnboundReaderReuse()
+        {
+            var csb = new NpgsqlConnectionStringBuilder(ConnectionString)
```

Comment:
For this kind of case, we usually use a temporary pool which gets destroyed at the end of the test; otherwise the pooled physical connection used here will be kept open for the entire test suite. Check out CreateTempPool.